### ISSUE TITLE
`./x.py test --pass check` fails

### ISSUE THREAD

**Issue by RalfJung** [rust-lang/rust#66929]
The following command fails on master:
```
./x.py -j4 test --stage 1 src/test/ui --test-args issue-65419 --pass check
```
says
```
[...]
---- [ui] ui/async-await/issues/issue-65419/issue-65419-async-fn-resume-after-panic.rs stdout ----

error: error pattern ' thread 'main' panicked at '`async fn` resumed after panicking'' not found!
status: exit code: 0
command: "/home/r/src/rust/rustc.2/build/x86_64-unknown-linux-gnu/stage1/bin/rustc" "/home/r/src/rust/rustc.2/src/test/ui/async-await/issues/issue-65419/issue-65419-async-fn-resume-after-panic.rs" "-Zthreads=1" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "-Zui-testing" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/home/r/src/rust/rustc.2/build/x86_64-unknown-linux-gnu/test/ui/async-await/issues/issue-65419/issue-65419-async-fn-resume-after-panic" "-Crpath" "-O" "-Cdebuginfo=0" "-Zunstable-options" "-Lnative=/home/r/src/rust/rustc.2/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "--edition=2018" "-L" "/home/r/src/rust/rustc.2/build/x86_64-unknown-linux-gnu/test/ui/async-await/issues/issue-65419/issue-65419-async-fn-resume-after-panic/auxiliary" "-A" "unused"
stdout:
------------------------------------------

------------------------------------------
stderr:
------------------------------------------

------------------------------------------




failures:
    [ui] ui/async-await/issues/issue-65419/issue-65419-async-fn-resume-after-completion.rs
    [ui] ui/async-await/issues/issue-65419/issue-65419-async-fn-resume-after-panic.rs
    [ui] ui/async-await/issues/issue-65419/issue-65419-generator-resume-after-completion.rs
```
Omitting the `--pass check` fixes that.  The offending tests are all `run-fail` tests with an `error-pattern`, looks like those don't work properly with `--pass check`?

Cc @Mark-Simulacrum

**Comment by petrochenkov** [comment]
cc @ninjasource https://github.com/rust-lang/rust/pull/66321
cc @nvlbg https://github.com/rust-lang/rust/pull/65506
Please, don't move `run-fail` tests to UI until this is addressed.

cc https://github.com/rust-lang/rust/issues/65865

**Comment by ninjasource** [comment]
Hi @petrochenkov, Should I create a PR to move the issue-65419 tests back to src/text/run-fail? Happy to do so.

**Comment by Centril** [comment]
I have a possible fix in #66932.

**Comment by ninjasource** [comment]
@Centril, great. Let me know if it doesn't work out and I'll submit my change as an alternative fix.
